__tests__/topics.test.js

```diff
@@ -305,7 +305,150 @@ describe("Articles api testing /api/articles ", () => {
+                        });
+                    });
+                });
+                }); //-- test
+
+        
+        
+        test(`sort articles by ${sort_by}`, () => {
+            order = 'desc'; // default
+            
+            // console.log(`"checking sort_by = ${sort_by} : order = ${order}`)            
+            return request(app)
+                .get(`/api/articles?sort_by=${sort_by}`) 
+                .expect(200)
+                .then(({ body }) => {
+                    const { articles } = body;
+                    
+                    if (order === 'asc'){
+                        expect(articles).toBeSortedBy( `${sort_by}`);
+                    }
+                    else {
+                        expect(articles).toBeSortedBy( `${sort_by}`, { descending: true });
+                    }
+                    
+                    articles.forEach((article) => {
+                        expect(article).toMatchObject({
+                            article_id: expect.any(Number),
+                            title: expect.any(String),
+                            topic: expect.any(String),
+                            author: expect.any(String),
+                            body: expect.any(String),
+                            created_at: expect.any(String),
+                            votes: expect.any(Number),
+                            comment_count: expect.any(Number),
+                        });
+                    });
+                });
+                }); //-- test
+
+        test(`sort articles by ${sort_by}`, () => {
+            // console.log(`"checking sort_by = ${sort_by} : order = ${order}`)            
+            return request(app)
+                .get(`/api/articles?sort_by=${sort_by}&order=${order}`) 
+                .expect(200)
+                .then(({ body }) => {
+                    const { articles } = body;
+                    
+                    if (order === 'asc'){
+                        expect(articles).toBeSortedBy( `${sort_by}`);
+                    }
+                    else {
+                        expect(articles).toBeSortedBy( `${sort_by}`, { descending: true });
+                    }
+                    
+                    articles.forEach((article) => {
+                        expect(article).toMatchObject({
+                            article_id: expect.any(Number),
+                            title: expect.any(String),
+                            topic: expect.any(String),
+                            author: expect.any(String),
+                            body: expect.any(String),
+                            created_at: expect.any(String),
+                            votes: expect.any(Number),
+                            comment_count: expect.any(Number),
+                        });
+                    });
+                });
+        }); //-- test
+        
+    });
+    });
+    });
+    
+}); // describe end
+
```

Comment:
Should have some tests for the following errors:
  - [ ] Status 400. invalid `sort_by` query, e.g. `?sort_by=bananas`
  - [ ] Status 400. invalid `order` query, e.g. `?order=bananas`
  - [ ] Status 404. non-existent `topic` query, e.g. `?topic=bananas`
  - [ ] Status 200. valid `topic` query, but has no articles responds with an empty array of articles, e.g. `?topic=paper`